src/App.tsx

```diff
@@ -2,28 +1,57 @@
+/* eslint-disable @typescript-eslint/indent */
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const TOTAL_ITEMS = 42;
+const items = getNumbers(1, TOTAL_ITEMS).map(n => `Item ${n}`);
+const itemsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(itemsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const itemsFrom = perPage * currentPage - perPage + 1;
+  const itemsTo = Math.min(perPage * currentPage, TOTAL_ITEMS);
+
+  const visibleItems = items.filter(
+    (_item, index) => index >= itemsFrom - 1 && index <= itemsTo - 1,
```

Comment:
Using the filter method to calculate visible items is not the most efficient way, as it will iterate over all items even if only a subset is needed. Consider using slice method instead, which can directly extract the sub-array of visible items based on `itemsFrom` and `itemsTo`.